src/control/client/connect_test.go

```diff
@@ -292,10 +305,83 @@ func TestKillRank(t *testing.T) {
 
 	for _, tt := range tests {
 		cc := connectSetup(log, Ready, MockFeatures, MockCtrlrs, MockCtrlrResults, MockModules,
-			MockModuleResults, MockPmemDevices, MockMountResults, nil, nil, nil, nil, tt.killRet, nil)
+			MockModuleResults, MockPmemDevices, MockMountResults, nil, nil, nil,
+			nil, tt.killRet, nil, MockACL)
 
 		resultMap := cc.KillRank("acd", 0)
 
 		checkResults(t, Addresses{MockServers[0]}, resultMap, tt.killRet)
 	}
 }
+
+func TestPoolGetACL(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer ShowBufferOnFailure(t, buf)()
+
+	tests := []struct {
```

Comment:
I recommend the pattern of map[string]struct{} for these table-driven tests, where the key is the description. This has the benefit of simplifying the struct as well as introducing a bit of randomness to the test execution as map iteration is nondeterministic.

Also, there's no reason to declare a standalone tests variable... You can declare the for loop and declare the map in one go, e.g.

```go
for name, tc := range map[string]struct{
   // struct def
}{
  "test one": {
    // test params
  },
}{
  t.Run(name, func(t *testing.T) {
    // run the subtest
  })
}
```

Finally, note that each test case can be run as a subtest using t.Run(). This improves output and makes it easier to understand exactly which test case failed when there is a failure.

Reply:
Massaged these tests to fit the recommended pattern. Thanks.

Reply:
Massaged the tests to fit the recommended pattern. Thanks.